crates/sui-archival/src/reader.rs

```diff
@@ -265,6 +266,80 @@ impl ArchiveReader {
             .await
     }
 
+    pub async fn load_summaries_upto<S>(&self, epoch: EpochId, store: S) -> Result<()>
```

Comment:
I wonder if it'll be better to pass a flag into the other read function to only load summaries  vs duplicating code here.

Reply:
I considered that at first, but ultimately decided not to, given that there are a number of small differences. I also may need to make some changes to this function to allow it to be `Send` so that I run it in a separate thread, and those may cause them to diverge further. But I could be convinced to refactor once i've made those changes